src/test_host.cpp

```diff
@@ -742,14 +743,20 @@ void TestHost::SaveBackBuffer(const std::string &output_directory, const std::st
   auto height = static_cast<int>(pb_back_buffer_height());
   auto pitch = static_cast<int>(pb_back_buffer_pitch());
 
-  SDL_Surface *surface =
-      SDL_CreateRGBSurfaceWithFormatFrom((void *)buffer, width, height, 32, pitch, SDL_PIXELFORMAT_ARGB8888);
-  if (IMG_SavePNG(surface, target_file.c_str())) {
-    PrintMsg("Failed to save PNG file '%s'\n", target_file.c_str());
-    ASSERT(!"Failed to save PNG file.");
-  }
+  ASSERT((pitch == width * 4) && "Encoder expects packed 24 or 32bpp");
 
-  SDL_FreeSurface(surface);
+  std::vector<uint8_t> out_buf;
+  if (!fpng::fpng_encode_image_to_memory((void*)buffer, width, height, 4, out_buf)) {
+    ASSERT(!"Failed to encode PNG image");
+  }
+  FILE *pFile = fopen(target_file.c_str(), "wb");
+  ASSERT(pFile && "Failed to open output PNG image");
+  if (fwrite(out_buf.data(), 1, out_buf.size(), pFile) != out_buf.size()) {
+    ASSERT(!"Failed to write output PNG image");
+  }
+  if (fclose(pFile)) {
+    ASSERT(!"Failed to close output PNG image");
+  }
```

Comment:
New encoder could be used below for depth buffer encoding, but its left for a follow-up